### PR TITLE
Add Florence2 model support and tests

### PR DESCRIPTION
**Pull request by houleisai**
```markdown
## PR 内容

本 PR 新增 Florence2 模型在 PaddleFormers 中的接入与验证，包括：

- 新增 Florence2 模型配置、建模、图像处理器和 Processor。
- 注册 Florence2 到 AutoConfig / AutoModel / AutoProcessor / AutoImageProcessor。
- 新增 Florence2 多模态 SFT template 和 mm plugin
- 新增 Florence2 SFT 训练配置示例。
- 新增 Florence2 模型单测和 Processor 单测。

## 验证结果

### 1. 单卡前向精度对齐
分别在 transformers/PyTorch 和 PaddleFormers 中进行前向对齐验证。

结果：

```text
logits mean abs diff: 0.00225268
logits p99 abs diff:  0.00976562
logits max abs diff:  0.02343750
first 10 generated tokens equal: True
```
结论：logits P99 diff 达到 `1e-2` 量级，且前 10 个生成 token 与 transformers 一致。

### 2. 生成结果验证

- transformers/PyTorch 端手动生成 10 个 token。
- PaddleFormers 端使用 `model.generate(..., max_new_tokens=10)`。
- 对比前 10 个生成 token。

结果：first 10 generated tokens equal: True

### 4. 新增模型单测

新增 Florence2 模型单测：

```bash
../miniconda3/envs/paddleformers/bin/python -m unittest \
tests.transformers.florence2.test_modeling
```

覆盖内容包括：

- Config 保存 / 加载 / 字段一致性
- forward + loss
- save/load 输出一致性
- determinism
- hidden states 输出
- resize token embeddings
- greedy / beam / sample generation
- SFT labels 拆分

新增 Florence2 Processor 单测：

```bash
../miniconda3/envs/paddleformers/bin/python -m unittest \
tests.transformers.florence2.test_processor
```

覆盖内容包括：

- task prompt 构造
- detection post process
- segmentation post process
- region proposal post process
- Florence2 多模态 SFT message 格式处理

### CODE REVIEW

File: paddleformers/transformers/florence2/modeling.py [hunk 1]
```diff
@@ -0,0 +1,942 @@
...
+        return outputs
+
+    def generate(self, input_ids=None, pixel_values=None, inputs_embeds=None, attention_mask=None, **kwargs):
+        if inputs_embeds is None:
+            inputs_embeds = self.get_input_embeddings()(input_ids)
+            if pixel_values is not None:
+                inputs_embeds, attention_mask = self._merge_image_features(
+                    self._encode_image(pixel_values), inputs_embeds, attention_mask
+                )
+        return self.language_model.generate(inputs_embeds=inputs_embeds, attention_mask=attention_mask, **kwargs)
```
Comment (risemeup1111) [inline paddleformers/transformers/florence2/modeling.py:977]: P1

公开的 `Florence2ForConditionalGeneration.generate()` 在 beam 扩展阶段会把 encoder 返回的 `BaseModelOutput` 直接交给 `paddle.gather`，因此 `decode_strategy="beam_search"` 会在首轮扩展时报类型错误。当前 `test_beam_search_generate` 手动取 `.last_hidden_state`、先重复 Tensor 再重新包装，并绕过 `model.generate()` 直接调用 `language_model.beam_search`，没有覆盖这条公共路径。请修正 encoder output 的 beam 扩展（保留输出对象并只扩展其中的 Tensor），并增加真实 `model.generate(..., num_beams=2)` 回归测试。

File: paddleformers/transformers/florence2/modeling.py [hunk 2]
```diff
@@ -0,0 +1,942 @@
...
+    def _split_sft_inputs(self, input_ids, labels, attention_mask):
+        source_rows, label_rows = [], []
+        max_source = 1
+        max_target = 1
+        for row, label_row in zip(input_ids.tolist(), labels.tolist()):
+            target_start = next((index for index, value in enumerate(label_row) if value != -100), len(row))
+            # PaddleFormers SFT labels are shifted left once, so the first
+            # supervised label predicts the token after this source position.
+            source = row[: target_start + 1] or [self.config.bos_token_id]
+            target = [value for value in label_row[target_start:] if value != -100]
```
Comment (risemeup1111) [inline paddleformers/transformers/florence2/modeling.py:908]: P1

`_split_sft_inputs` 只把第一个非 `-100` label 之前的一段作为 source，却把后续所有非 `-100` label 拼成一个 target。多轮 `SFTDataset` 会在各 assistant span 之间保留下一轮 user token 并把这些位置标为 `-100`；例如 `[10,11,20,21,12,13,30,31,2]` 会被拆成 source `[10,11]`、target `[20,21,30,31,2]`，第二轮上下文 `[12,13]` 被静默丢弃。请明确拒绝多轮 Florence 样本，或按每个 supervised turn 拆成独立 seq2seq 样本，并补充多轮 labels 的回归测试。

File: examples/config/sft-vl/florence2_full_300_steps.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+eval_dataset_path: ./florence2_train.jsonl
+eval_dataset_prob: "1.0"
+max_seq_len: 1024
+packing: false
+mix_strategy: concat
+template_backend: custom
+template: florence2
+
+### model
+model_name_or_path: /home/housaijie/code/Florence-2-base
```
Comment (risemeup1111) [inline examples/config/sft-vl/florence2_full_300_steps.yaml:18]: P2

示例配置把 `model_name_or_path` 固定为作者机器上的 `/home/housaijie/code/Florence-2-base`，其他开发者或 CI 无法直接运行；同目录的其他配置使用可移植的模型 ID。请改为 `microsoft/Florence-2-base`（或明确的可配置占位符），并保留 `convert_from_hf: true`。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
| 序号 | 位置 | 优先级 | 状态 |
| --- | --- | --- | --- |
| 1 | [beam search 生成路径](https://github.com/PaddlePaddle/PaddleFormers/pull/4612#discussion_r3755571547) | P1 | ✅ |
| 2 | [SFT 多轮标签拆分](https://github.com/PaddlePaddle/PaddleFormers/pull/4612#discussion_r3755574849) | P1 | 🚧 |
| 3 | [Florence-2 示例模型路径](https://github.com/PaddlePaddle/PaddleFormers/pull/4612#discussion_r3755576508) | P2 | 🚧 |

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: paddleformers/transformers/florence2/modeling.py (hunk 1 above)
Comment (risemeup1111) [reply paddleformers/transformers/florence2/modeling.py:977]: 已在当前 head 的 `paddleformers/transformers/florence2/modeling.py:770-803` 复核：新增 override 会保留 `BaseModelOutput` 并按 beam index 扩展其 Tensor；`tests/transformers/florence2/test_modeling.py:211-218` 也已改为直接调用 `model.generate(..., decode_strategy="beam_search", num_beams=2)`。原 beam search P1 已有对应修复，我将关闭此 canonical thread。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
